lms/services/application_instance.py

```diff
@@ -52,14 +52,14 @@ def __init__(self, application_instance: ApplicationInstance):
         )
 
 
-def _email_or_domain_match(columns, email):
+def _email_or_domain_match(columns, email: str):
     """
     Get an SQL comparator for matching emails.
 
     This will match the full email if it contains '@' or interpret the text
     as a domain if not. This will search over all the provided fields.
     """
-    return sa.or_(
+    return sa.or_(  # type: ignore
```

Comment:
What is the issue here? Also, can you add a comments alongside the `type: ignore`? In the frontend we have markers like this:

```
 // @ts-ignore - TS doesn't know about PDFViewerApplication global.
```

As TS fortunately allows us to add notes for future readers.

Reply:
One other useful thing TS can do is to indicate that you _expect_ an error:

```
 // @ts-expect-error - Navigation API is missing from TS
```

This will cause a warning or error if the suppression subsequently becomes unnecessary. Is there a mechanism in mypy to inform us if a `# type: ignore` becomes unnecessary?

Reply:
>  Is there a mechanism in mypy to inform us if a # type: ignore becomes unnecessary?

Yes, we'll get an error for those with the current conf :+1: